lib/UmdMainTemplatePlugin.js

```diff
@@ -7,22 +7,47 @@
 const { ConcatSource, OriginalSource } = require("webpack-sources");
 const Template = require("./Template");
 
-function accessorToObjectAccess(accessor) {
+/** @typedef {import("./Compilation")} Compilation */
+
+/**
+ * @param {string[]} accessor the accessor to convert to path
+ * @returns {string} the path
+ */
+const accessorToObjectAccess = accessor => {
 	return accessor.map(a => `[${JSON.stringify(a)}]`).join("");
-}
+};
 
-function accessorAccess(base, accessor) {
-	accessor = [].concat(accessor);
-	return accessor
-		.map((a, idx) => {
-			a = base + accessorToObjectAccess(accessor.slice(0, idx + 1));
-			if (idx === accessor.length - 1) return a;
+/**
+ * @param {string=} base the path prefix
+ * @param {string|string[]} accessor the accessor
```

Comment:
Same problem here as in LibraryTemplatePlugin

(actually, shouldn't these functions be put in a common module?)